packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -345,6 +346,61 @@ private static void GetParamInfo(Dictionary<string, ParameterProvider> paramMap,
             }
         }
 
+        private static IReadOnlyList<ParameterProvider> BuildOrderedSpreadParameters(InputModelType inputModel)
+        {
+            // create the model provider
+            if (ClientModelPlugin.Instance.TypeFactory.CreateModel(inputModel) is not ModelProvider modelProvider)
```

Comment:
I am not sure we want to put this type restriction on here, if we do, then we should change the type factory return type and require that everyone inherit from ModelProvider.

@JoshLove-msft  thoughts?

Reply:
well this is my initial suggestion - if some one wants to change something in the models they should override our models so that they do not have to recursively change everything downstream.
They could also change everything in our modelprovider by inherit modelprovider - exactly the same as they build everything from scratch (TypeProvider) because we never add `sealed` to anything

Reply:
Resolving this as I removed the dep. on modelprovider and instead rely on a provider's properties to build the parameters in the case of spread.